### PR TITLE
adds lightbulb removing skillchip

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

adds lightbulb removing skillchip to the fun and games vendor, it gives you the skill to remove a light bulb without gloves after 5 seconds, burning your hand a lot and shattering the bulb 
https://streamable.com/qgpqal

## Why It's Good For The Game

Consider the simple act of trade. If you believe that removing a light bulb would increase your happiness, and if my happiness would be increased by selling that skill to you for a certain sum, then the trade makes us both happy, and the amount of happiness in the universe is increased.

## Changelog
:cl:
add: lightbulb removing skillchip
/:cl:

### CODE REVIEW

**Comment by Ryll-Ryll** [comment]
didn't take long for powercreep to set in huh

**Comment by CRITAWAKETS** [comment]
what about prisoners in the permabrig or gulag, they could remove their lightbulbs, dont you see how much this BREAKS balance?

**Comment by ATH1909** [comment]
You could probably buff this to allow you to safely extract lightbulbs with your bare hands without taking burn damage or damaging the light bulb, and it still wouldn't be too out of line.

**Comment by MrDoomBringer** [comment]
this is a ryll-tier PR good work

**Comment by Fikou** [comment]
> You could probably buff this to allow you to safely extract lightbulbs with your bare hands without taking burn damage or damaging the light bulb, and it still wouldn't be too out of line with the other joke skillchips (you could just smash the light bulb during those 5 seconds instead of extracting it).

its a skillchip, it modifies your nerves with your brain to not feel the pain of having your hand burn from taking the light bulb so you can pull it off, it doesnt stop the damage to your skin

**Comment by ATH1909** [comment]
> > You could probably buff this to allow you to safely extract lightbulbs with your bare hands without taking burn damage or damaging the light bulb, and it still wouldn't be too out of line with the other joke skillchips (you could just smash the light bulb during those 5 seconds instead of extracting it).
> 
> its a skillchip, it modifies your nerves with your brain to not feel the pain of having your hand burn from taking the light bulb so you can pull it off, it doesnt stop the damage to your skin

I thought it just taught you how to properly remove a lightbulb.

**Comment by Tlaltecuhtli** [comment]
you can use this for a certain thing not sure if intended

**Comment by Fikou** [comment]
> you can use this for a certain thing not sure if intended

what

File: code/modules/library/skill_learning/skillchip.dm [hunk 1]
```diff
@@ -97,15 +97,15 @@
 	skill_description = "Trim hedges and potted plants into marvelous new shapes with any old knife. Not applicable to plastic plants."
 	skill_icon = "spa"
 	implanting_message = "<span class='notice'>Your mind is filled with plant arrangments.</span>"
-	removal_message = "<span class='notice'>Your can't remember how a hedge looks like anymore.</span>"
+	removal_message = "<span class='notice'>You can't remember how a hedge looks like anymore.</span>"
```
Comment (ATH1909) [inline code/modules/library/skill_learning/skillchip.dm:100]:
```suggestion
	removal_message = "<span class='notice'>You can't remember what a hedge looks like anymore.</span>"
```

**Comment by Rohesie** [comment]
Looks like a janitor ability.

**Comment by Fikou** [comment]
> Looks like a janitor ability.

janitors already have black gloves and its a memeish ability

**Comment by Rohesie** [comment]
> 
> 
> > Looks like a janitor ability.
> 
> janitors already have black gloves and its a memeish ability

Can't you use this to escape from a regular brig cell? Maybe that's no longer a thing since the windows were reinforced.

**Comment by Fikou** [comment]
> > > Looks like a janitor ability.
> > 
> > 
> > janitors already have black gloves and its a memeish ability
> 
> Can't you use this to escape from a regular brig cell? Maybe that's no longer a thing since the windows were reinforced.

it has 2 force, rwindows and windoors have defense up to like 12 force

**Comment by Rohesie** [comment]
> 
> 
> > > > Looks like a janitor ability.
> > > 
> > > 
> > > janitors already have black gloves and its a memeish ability
> > 
> > 
> > Can't you use this to escape from a regular brig cell? Maybe that's no longer a thing since the windows were reinforced.
> 
> it has 2 force, rwindows and windoors have defense up to like 12 force

Fair enough.

**Comment by wesoda25** [comment]
talk about bikeshedding

**Comment by trollbreeder** [comment]
How many skillchips do you need to change a lightbulb? One to screw into your brainy bits, and a trip to the medbay after burning yourself on the bulb.

**Comment by Rohesie** [comment]
Skillchips have been refactored.
Make this work under the new system and I'll merge it.
The concept is approved.

File: code/modules/power/lighting.dm [hunk 2]
```diff
@@ -676,12 +676,20 @@
...
-
 			var/obj/item/bodypart/affecting = H.get_bodypart("[(user.active_hand_index % 2 == 0) ? "r" : "l" ]_arm")
 			if(affecting && affecting.receive_damage( 0, 5 ))		// 5 burn damage
 				H.update_damage_overlays()
-			return				// if burned, don't remove the light
+			if(HAS_TRAIT(user, TRAIT_LIGHTBULB_REMOVER))
+				to_chat(user, "<span class='notice'>You feel like you're burning, but you can push through.</span>")
+				if(!do_after(user, 50, target = src))
+					return
+				if(affecting && affecting.receive_damage( 0, 10 ))		// 10 more burn damage
```
Comment (Rohesie) [inline code/modules/power/lighting.dm:686]:
```suggestion
				if(affecting?.receive_damage( 0, 10 ))		// 10 more burn damage
```

File: code/modules/power/lighting.dm [hunk 3]
```diff
@@ -676,12 +676,20 @@
...
 		else
-			to_chat(user, "<span class='warning'>You try to remove the light [fitting], but you burn your hand on it!</span>")
-
 			var/obj/item/bodypart/affecting = H.get_bodypart("[(user.active_hand_index % 2 == 0) ? "r" : "l" ]_arm")
 			if(affecting && affecting.receive_damage( 0, 5 ))		// 5 burn damage
 				H.update_damage_overlays()
-			return				// if burned, don't remove the light
+			if(HAS_TRAIT(user, TRAIT_LIGHTBULB_REMOVER))
+				to_chat(user, "<span class='notice'>You feel like you're burning, but you can push through.</span>")
+				if(!do_after(user, 50, target = src))
```
Comment (Rohesie) [inline code/modules/power/lighting.dm:684]:
```suggestion
				if(!do_after(user, 5 SECONDS, target = src))
```

**Pull request merged**